lib/Sema/CSSolver.cpp

```diff
@@ -2421,7 +2421,26 @@ bool ConjunctionElement::attempt(ConstraintSystem &cs) const {
     for (auto *typeVar : referencedVars)
       cs.addTypeVariable(typeVar);
   }
-
+  // Let's try calling precheck again even if that means modifying the AST twice
```

Comment:
We don't want to call `preCheckExpression` twice - we want to call it only once, but call it here before the `SyntacticElement` constraint is solved (and constraints are generated for the AST node).